### PR TITLE
Add support for dynamic array serialization

### PR DESCRIPTION
**Pull request by oleksiivorobiov**

### CODE REVIEW

File: MongoBsonSerializer.pas [hunk 1]
```diff
@@ -305,6 +305,8 @@ procedure TPrimitivesBsonSerializer.SerializePropInfo(APropInfo: PPropInfo);
       end;
     tkClass : SerializeObject(APropInfo);
     tkVariant : SerializeVariant(APropInfo, APropInfo.Name, Null);
+    tkDynArray :
+      SerializeVariant(nil, APropInfo.Name, GetPropValue(Source, APropInfo));
```
Comment (jsbattig) [inline MongoBsonSerializer.pas:309]: How deserialization works?
Is it implemented in Pull #31 ?

Comment (oleksiivorobiov) [inline MongoBsonSerializer.pas:309]: No. There was no dynamic arrays support(just variant) as we can get property as variant we can just use already written function.

Comment (jsbattig) [inline MongoBsonSerializer.pas:309]: Alex, didn't work if you did something like this:

``` pascal
tkVariant, tkDynArray  : SerializeVariant(APropInfo, APropInfo.Name, Null);
```

Comment (oleksiivorobiov) [inline MongoBsonSerializer.pas:309]: Yes it didn't in this case.

**Pull request merged**